### DISCUSSION TITLE
Install a library for all the virtualenvs

### DISCUSSION THREAD

**Discussion by lcfd** [python-poetry/poetry#6771] [Q&A] [answered]
Hi!

I'm trying to use `pynvim` to add Python support to Nvim. So I need this package always available.

So I installed the library:
`PIP_REQUIRE_VIRTUALENV=false pip install pynvim`

The package is available when I'm not in a Poetry venv:
`/Users/xxx/.asdf/installs/python/3.10.6/lib/python3.10/site-packages/pynvim/__init__.py`

When I'm in a Poetry environment `pynvim` is not available anymore.

I tried using this configuration:
`poetry config virtualenvs.options.system-site-packages true`

But the package is still not available. And it doesn't show up in the `poetry show` list.

Any idea about what I should do?

**Comment by neersighted** [comment, marked as answer]
You want neovim to always use the same virtual environment, regardless of the virtual environment it is within, in my experience. To do so, you should create a virtual environment to install pynvim in:

```sh
python3 -m venv ~/.config/nvim/venv
~/.config/nvim/venv/bin/pip install pynvim
```

And then configure Vim to always use it:
```viml
let g:python3_host_prog = "~/.config/nvim/venv/bin/python3"
```

I'm a little rusty on specifics, but hopefully that puts you on the right track.

**Reply by lcfd** [reply]
> but hopefully that puts you on the right track

Hi! Absolutely! Thank you for the suggestion.

I am a little surprised that I am the first that needs such a thing.
Let's see if anyone else has any other ideas.

So Neovim a part, it's not possible to use a global package alongside Poetry?

**Reply by neersighted** [reply]
If you want something in your Poetry environment, it needs to be in pyproject.toml. Poetry does not have a feature where it installs packages globally -- such would be counter to the goal of consistent and reproducible installations.

**Reply by lcfd** [reply]
> such would be counter to the goal of consistent and reproducible installations.

Sure, at the moment your solution should be enough.
